### ISSUE TITLE
Parallelize the IO for files

### ISSUE THREAD

**Issue by elpiel** [rzip/rzip#9]
Maybe we can use this:
https://docs.rs/async-std/0.99.11/async_std/io/index.html

Rather than having `unzip_archive`, we can collect the `ZipFile`s in 2 different `Vec`s, directories and files, create the directories (maybe also async) and then write the files, again async. 

**Note:** I am not sure if it shows all directories when you list them as I see you use `create_dir_all`, but maybe it's because we cannot be sure if the directory already exists, because of the ordering of files inside

This is just an idea, I want to hear your thoughts. I will try though it and probably open a WIP PR.

**Comment by DavidBM** [comment]
As far I know, only a ZipFile can exist in parallel. That means, it works with iterators when you use FnOnce and for lops, but you can't store them and you can't have parallel iterators like rayon. 

You can check here: https://github.com/mvdnes/zip-rs/issues/14
The last comment before mine and the pull requests are interesting. Still, no movement in that regard by the library. :(

**Comment by elpiel** [comment]
Indeed, you are right. It will prove to be a bit too much work right now.
Let's leave it for now and after that we can even make a fork and go async there as well. Might be just using `async_std` initially, might be doing more, but not now.
